host/hero-target.c

```diff
@@ -96,3 +96,27 @@ hero_rt_core_id(void)
 {
   return omp_get_thread_num();
 }
+
+void
+hero_rt_start_cycle_cnt()
+{
+	return;
```

Comment:
Shouldn't there be a corresponding timer implementation for the host side as well? I think this would be useful when doing profiling and comparing host vs. accelerator execution. What do you think?

Reply:
The PULP timer returns the number of cycles. How would you implement this on the host while being consistent in the unit (clock cycles)?

Reply:
I see two options:
1. You are using functions like `clock_gettime()` on the host which gives you the time in seconds and nanoseconds. You can then convert this into accelerator clock cycles or host clock cycles depending on what comparison you want to do. The frequencies of these two clocks you can get from the `pulp` struct defined and set up by libpulp (see hero-support), namely `pulp->pulp_clk_freq_mhz` and `pulp->host_clk_freq_mhz`. Alternatively, you can also get the host clock frequency from sysfs.

2. Despite executing on the host, you can still use the timer inside the accelerator directly. How to do this is illustrated here:
https://github.com/pulp-platform/hero-support/blob/6762d055953089eea215c60a58ad3a62c6b717e2/libpulp/src/pulp.c#L503
You just need to be careful that the accelerator is not manipulating the timer simultaneously. 

Best regards,
Pirmin